IdentityCore/src/MSIDAccount.m

```diff
@@ -78,7 +248,52 @@ - (NSString *)userIdentifier
     {
         return [NSString stringWithFormat:@"%@.%@", self.uid, self.utid];
     }
-    return nil;    
+    return self.uniqueUserId;
+}
+
+#pragma mark - Helpers
+
+- (NSString *)accountTypeString
+{
+    switch (self.accountType)
+    {
+        case MSIDAccountTypeAADV1:
+            return @"AAD";
+            
+        case MSIDAccountTypeMSA:
+            return @"MSA";
+            
+        case MSIDAccountTypeAADV2:
+            return @"MSSTS";
+            
+        default:
+            return @"Other";
+    }
+}
+
+static NSDictionary *accountTypes = nil;
```

Comment:
'accountTypes' is not used outside of accountTypeFromString: Can we move it as local variable to accountTypeFromString:, or even better -- use dispatch_once?